src/Plugin/jsonapi/FieldEnhancer/CardLinkEnhancer.php

```diff
@@ -136,25 +141,27 @@ public function getCardFields(EntityInterface $node) {
           $event_date = $paragraph->field_paragraph_date_range->getValue()[0];
           // Parse date with GMT timezone.
           if ($event_date != NULL) {
-            $storage_tz = DateTimeItemInterface::STORAGE_TIMEZONE;
-            $date_value = new DrupalDateTime($event_date['value'], $storage_tz);
-            $date_end_value = new DrupalDateTime($event_date['end_value'], $storage_tz);
-            $system_tz = \Drupal::service('config.factory')->get('system.date')->get('timezone.default');
-            // Convert to local timezone.
-            $date_formatter = \Drupal::service('date.formatter');
-            $event_date = [
-              'value' => $date_formatter->format($date_value->getTimeStamp(), 'custom', 'Y-m-d H:i:s', $system_tz),
-              'end_value' => $date_formatter->format($date_end_value->getTimeStamp(), 'custom', 'Y-m-d H:i:s', $system_tz),
+            $start_date = $event_date['value'] ?? NULL;
+            $end_date = $event_date['end_value'] ?? NULL;
+            $event_dates = [
+              'value' => TideLandingPageHelper::localDateAndTimeFormatter($start_date),
+              'end_value' => TideLandingPageHelper::localDateAndTimeFormatter($end_date),
             ];
+            $card_fields['date'] = $event_dates ? $event_dates : '';
           }
-          $card_fields['date'] = $event_date ? $event_date : '';
         }
       }
     }
     if ($module_handler->moduleExists('tide_grant')) {
       // Add the date field for grants.
       if ($node->hasField('field_node_dates') && !$node->field_node_dates->isEmpty()) {
-        $card_fields['date'] = $node->get('field_node_dates')->getValue()[0];
+        $start_date = $node->get('field_node_dates')->getValue()[0]['value'] ?? NULL;
+        $end_date = $node->get('field_node_dates')->getValue()[0]['end_value'] ?? NULL;
+        $grants_dates = [
+          'value' => TideLandingPageHelper::localDateAndTimeFormatter($start_date),
+          'end_value' => TideLandingPageHelper::localDateAndTimeFormatter($end_date),
+        ];
+        $card_fields['date'] = $grants_dates ? $grants_dates : '';
```

Comment:
If you have good tests on the helper, this reduces the fear that the rest of the code goes wrong because it is literally just making the helper do the heavy lifting.

Reply:
Further to this, it's the _reason_ it's in a helper :). So it can be unit tested.